dipy/tracking/utils.py

```diff
@@ -420,6 +431,75 @@ def seeds_from_mask(mask, density=[1, 1, 1], voxel_size=None, affine=None):
 
     return seeds
 
+def random_seeds_from_mask(mask, seeds_per_voxel=1, voxel_size=None, affine=None):
+    """Creates randomly placed seeds for fiber tracking from a binary mask.
+
+    Seeds points are placed randomly distributed in all voxels of ``mask`` which
+    are ``True``. This function is essentially similar to ``seeds_from_mask()``,
+    with the difference that instead of evenly distributing the seeds, it randomly
+    places the seeds within the voxels specified by the ``mask``
+
+    Parameters
+    ----------
+    mask : binary 3d array_like
+        A binary array specifying where to place the seeds for fiber tracking.
+    seeds_per_voxel : int
+        Specifies the number of seeds to place in each voxel.
+    voxel_size :
+        This argument is deprecated.
+    affine : array, (4, 4)
+        The mapping between voxel indices and the point space for seeds. A
+        seed point at the center the voxel ``[i, j, k]`` will be represented as
+        ``[x, y, z]`` where ``[x, y, z, 1] == np.dot(affine, [i, j, k , 1])``.
+
+    See Also
+    --------
+    seeds_from_mask
+
+    Raises
+    ------
+    ValueError
+        When ``mask`` is not a three-dimensional array
+
+    Examples
+    --------
+    >>> mask = np.zeros((3,3,3), 'bool')
+    >>> mask[0,0,0] = 1
+    >>> random_seeds_from_mask(mask, seeds_per_voxel=1, voxel_size=[1,1,1])
+    TODO
+
+    >>> random_seeds_from_mask(mask, [1,2,3], [1,1,1])
+    TODO
+    >>> mask[0,1,2] = 1
+    >>> random_seeds_from_mask(mask, [1,1,2], [1.1,1.1,2.5])
+    TODO
+
+    """
+    mask = np.array(mask, dtype=bool, copy=False, ndmin=3)
+    if mask.ndim != 3:
+        raise ValueError('mask cannot be more than 3d')
+
+    where = np.argwhere(mask)
+    num_voxels = len(where)
+
+    # Generate as many random triplets as the number of seeds needed
+    grid = np.random.random([seeds_per_voxel * num_voxels, 3])
+    # Repeat elements of 'where' so that it can be added to grid
+    where = np.repeat(where, seeds_per_voxel, axis=0)
+    seeds = where + grid - .5
+    seeds = asarray(seeds)
+
+    # Apply the spacial transform
+    if affine is not None:
+        # Use affine to move seeds int real world coordinates
+        seeds = np.dot(seeds, affine[:3, :3].T)
+        seeds += affine[:3, 3]
+    elif voxel_size is not None:
```

Comment:
We don't need to worry about voxel size here, we left it in the other functions for backwards compatibility, but we don't need to add it to new functions.